src/ReCaptcha.php

```diff
@@ -368,85 +354,93 @@ public function setPrivateKey($privateKey)
      *
      * This method uses the public key to fetch a recaptcha form.
      *
-     * @param null|string $name Base name for recaptcha form elements
+     * @param null|string $name
+     *            Base name for recaptcha form elements
      * @return string
-     * @throws Exception
+     * @throws \ZendService\ReCaptcha\Exception
      */
-    public function getHtml($name = null)
+    public function getHtml()
     {
-        if ($this->publicKey === null) {
-            throw new Exception('Missing public key');
+        if ($this->siteKey === null) {
+            throw new Exception('Missing site key');
         }
 
         $host = self::API_SERVER;
 
-        if ((bool) $this->params['ssl'] === true) {
-            $host = self::API_SECURE_SERVER;
-        }
-
-        $htmlBreak = '<br>';
-        $htmlInputClosing = '>';
-
-        if ((bool) $this->params['xhtml'] === true) {
-            $htmlBreak = '<br />';
-            $htmlInputClosing = '/>';
+        // Should We use a onload callback?
+        if (!empty($this->options['onload'])) {
+            return sprintf(
+                '<script type="text/javascript" src="%s.js?onload=%s&render=explicit" async defer></script>',
+                $host,
+                $this->options['onload']
+            );
         }
 
-        $errorPart = '';
+        $langOption = '';
 
-        if (!empty($this->params['error'])) {
-            $errorPart = '&error=' . urlencode($this->params['error']);
+        if (!empty($this->options['hl'])) {
+            $langOption = "?hl={$this->options['hl']}";
         }
 
-        $reCaptchaOptions = '';
-
-        if (!empty($this->options)) {
-            $encoded = \Zend\Json\Json::encode($this->options);
-            $reCaptchaOptions = <<<SCRIPT
-<script type="text/javascript">
-    var RecaptchaOptions = {$encoded};
-</script>
-SCRIPT;
-        }
-        $challengeField = 'recaptcha_challenge_field';
-        $responseField  = 'recaptcha_response_field';
-        if (!empty($name)) {
-            $challengeField = $name . '[' . $challengeField . ']';
-            $responseField  = $name . '[' . $responseField . ']';
+        $data = "data-sitekey=\"{$this->siteKey}\"";
+
+        foreach ([
+            'theme',
+            'type',
+            'size',
+            'tabindex',
+            'callback',
+            'expired-callback'
+        ] as $option) {
+            if (!empty($this->options[$option])) {
+                $data .= " data-$option=\"{$this->options[$option]}\"";
+            }
         }
 
-        $return = $reCaptchaOptions;
-        $return .= <<<HTML
-<script type="text/javascript"
-   src="{$host}/challenge?k={$this->publicKey}{$errorPart}">
-</script>
+        $return = <<<HTML
+<script type="text/javascript" src="{$host}.js{$langOption}" async defer></script>
+<div id="recaptcha_widget" class="g-recaptcha" $data></div>
```

Comment:
is it possible to use classes instead of ids? Using hard-coded html ids prevents multiple instances from working properly on a page.